Modules/timemodule.c

```diff
@@ -427,11 +427,12 @@ gettmarg(PyObject *args, struct tm *p)
         return 0;
     }
 
-    if (!PyArg_ParseTuple(args, "iiiiiiiii",
+    if (!PyArg_ParseTuple(args, "iiiiiiiii:mktime/asctime/strftime",
```

Comment:
indeed it is also related to bpo-28261 (I added a comment to that issue), but ISTM that it is also related to this issue, as currently the error messages are inconsistent:
```
>>> time.mktime((-1, 2,3,4,5,6,7,8,9))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OverflowError: mktime argument out of range
>>> time.mktime((-1 << 1000, 2,3,4,5,6,7,8,9))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OverflowError: Python int too large to convert to C long
```
the patches change it to:
```
>>> time.mktime((-1 ,2,3,4,5,6,7,8,9))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OverflowError: mktime argument out of range
>>> time.mktime((-1 << 1000 ,2,3,4,5,6,7,8,9))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OverflowError: mktime/asctime/strftime() argument out of range
```
isn't creating a local string, writing "iiiiiiiii" to it (maybe by calling strcpy()), and then using strcat() to concatenate the function name (received as an argument) a bit of an overkill?
(of course, if you think it is worth it, I will change it accordingly.)